core/src/main/java/org/apache/calcite/rel/metadata/RelMetadataQuery.java

```diff
@@ -537,6 +537,9 @@ public Boolean areColumnsUnique(RelNode rel, ImmutableBitSet columns) {
    */
   public Boolean areColumnsUnique(RelNode rel, ImmutableBitSet columns,
       boolean ignoreNulls) {
+    if (columns.length() > rel.getInputs().size()) {
```

Comment:
Would you please add a comment describing an edge case?
From a first glance it is not clear why the condition is like that.